mail/dovecot/Portfile

```diff
@@ -63,16 +64,23 @@ subport dovecot-sieve {
     homepage        https://pigeonhole.dovecot.org/
 
     distname        dovecot-${core_branch}-pigeonhole-${version}
-    master_sites    ${homepage}releases/${core_branch}/
 
     checksums       rmd160  d453de41682ce7c809cabb3c5ebddbe5e31eb804 \
                     sha256  637709a83fb1338c918e5398049f96b7aeb5ae00696794ed1e5a4d4c0ca3f688 \
                     size    1954347
 
     depends_lib-append \
-                    port:dovecot
+                    port:dovecot \
+                    port:gettext \
+                    port:mercurial
 
-    patchfiles      dynamic_lookup-11-sieve.patch
+    # Note: https://trac.macports.org/ticket/58544#comment:11 for this build logic
+    use_autoreconf  yes
+    autoreconf.cmd  ./autogen.sh
+    depends_build-append \
+                    port:autoconf \
+                    port:automake \
+                    port:libtool
```

Comment:
The small patchfile is IMO preferable to all this extra complexity.

Reply:
The complexity of the full distribution is necessary to be able to patch the code elsewhere. Please see e.g. https://github.com/macports/macports-ports/blob/master/mail/dovecot/files/dovecot-core-pushnotify.patch

Reply:
> Overall, this is reverting changes that were made deliberately and work fine as far as I can tell.

Please see how these change break the build at https://trac.macports.org/ticket/65712:
```sh
:info:build Undefined symbols for architecture x86_64:
:info:build   "_cmd_x_apple_push_service", referenced from:
:info:build Undefined symbols for architecture x86_64:
:info:build   "_cmd_x_apple_push_service", referenced from:
:info:build       _imap_ext_commands in imap-commands.o
:info:build       _imap_ext_commands in imap-commands.o
:info:build ld: symbol(s) not found for architecture x86_64
:info:build ld: symbol(s) not found for architecture x86_64
:info:build clang: error: linker command failed with exit code 1 (use -v to see invocation)
:info:build clang: error: linker command failed with exit code 1 (use -v to see invocation)
:info:build make[3]: *** [imap] Error 1
:info:build make[3]: *** Waiting for unfinished jobs....
:info:build make[3]: *** [test-imap-client-hibernate] Error 1
:info:build make[3]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_mail_dovecot/dovecot/work/dovecot-2.3.19.1/src/imap'
:info:build make[2]: *** [all-recursive] Error 1
:info:build make[2]: Leaving directory `/opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_mail_dovecot/dovecot/work/dovecot-2.3.19.1/src'
:info:build make[1]: *** [all-recursive] Error 1
```

Reply:
OK, so the apns variant is patching Makefile.am and configure.ac. The nicest options would be to patch Makefile.in and configure instead, but if that's too much work, the autoreconf and associated dependencies can be enabled in the apns variant only.